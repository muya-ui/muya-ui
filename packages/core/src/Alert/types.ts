import React from 'react';

import { IAnimationBaseProps } from '../Animation/types';
import { ICustomStyleBaseProps, ISizeSpecBaseProps } from '../types';

/**
 * Alert style keys
 * @styles wrapper Alert 外部容器
 * @styles container Alert 居中容器
 * @styles icon Icon 节点
 * @styles title 标题节点
 * @styles desc 描述节点
 * @styles close 关闭 Icon 节点
 */
export type IAlertStyleKeys = 'wrapper' | 'container' | 'icon' | 'title' | 'desc' | 'close';

type PropsFromAnimation = Pick<IAnimationBaseProps, 'appear' | 'timeout'>;

export interface IAlertPureProps
  extends ICustomStyleBaseProps<IAlertStyleKeys>,
    PropsFromAnimation {
  /**
   * 显示关闭按钮
   * @default false
   */
  showClose?: boolean;
  /**
   * 标题
   */
  title?: React.ReactNode;
  /**
   * 描述
   */
  description?: React.ReactNode;
  /**
   * 显示图标
   * @default true
   */
  showIcon?: boolean;
  /**
   * 类型
   * success-成功 error-失败 warning-警告 info-提示
   * @default 'info'
   */
  type?: 'success' | 'error' | 'warning' | 'info';
  /**
   * 显示与否，注意传了这个值 Alert 就变成受控组件了
   * @default true
   */
  visible?: boolean;
  /**
   * 点击关闭后的回调
   */
  onClose?: () => void;
  /**
   * children 当成是 title 使用
   */
  childrenAsTitle?: boolean;
  /**
   * 过渡组件
   * @default Animation.Collapse
   **/
  TransitionComponent?: React.ComponentType<any>;
}

export type IAlertProps = IAlertPureProps &
  ISizeSpecBaseProps &
  React.HTMLAttributes<HTMLDivElement>;
