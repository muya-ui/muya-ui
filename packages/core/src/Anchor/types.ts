import React from 'react';
import { ITabsProps, ITabProps } from '../Tabs';
import { Omit } from '../types';
import { IScrollViewProps } from '../ScrollView';

export type IAnchorContainer = HTMLElement | Window;

export interface IAnchorProviderProps {
  /**
   * 子节点
   *
   * @type {React.ReactNode}
   * @memberof IAnchorProviderProps
   */
  children?: React.ReactNode;
  /**
   * 距离窗口顶部达到指定偏移量后触发
   *
   * @type {number}
   * @memberof IAnchorProviderProps
   */
  offsetTop?: number;
  /**
   * 禁用后点击锚点不会修改location.hash
   *
   * @type {boolean}
   * @memberof IAnchorProviderProps
   */
  disableHash?: boolean;
  /**
   * 滚动动画过渡时间
   *
   * @default 300
   * @type {number}
   * @memberof IAnchorProviderProps
   */
  duration?: number;
  /**
   * 锚点的方向
   *
   * @default 'vertical'
   * @type {('vertical' | 'horizontal')}
   * @memberof IAnchorProviderProps
   */
  direction?: 'vertical' | 'horizontal';
  /**
   * 锚点区域边界
   *
   * @default 5
   * @type {number}
   * @memberof IAnchorProviderProps
   */
  bounds?: number;
  /**
   * 锚点变化事件
   *
   * @memberof IAnchorProviderProps
   */
  onChange?: (currentActiveLink: string) => void;
}

export type IAnchorScrollViewProps = IScrollViewProps;
export type IAnchorTabsProps = ITabsProps;
export interface IAnchorTabProps extends Omit<ITabProps, 'title'> {
  /**
   * 锚点索引，格式为`#{dom-id}`
   *
   * @type {string}
   * @memberof IAnchorTabProps
   */
  index?: string;
  /**
   * 锚点内容
   *
   * @type {React.ReactNode}
   * @memberof IAnchorTabProps
   */
  title?: React.ReactNode;
}

export interface IAnchorBag {
  /**
   * 当前激活的锚点
   *
   * @type {(string | null)}
   * @memberof IAnchorBag
   */
  activeLink: string | null;
  /**
   * 同 IAnchorProviderProps.hash
   *
   * @type {boolean}
   * @memberof IAnchorBag
   */
  disableHash: boolean;
  /**
   * 同 IAnchorProviderProps.direction
   *
   * @type {('vertical' | 'horizontal')}
   * @memberof IAnchorBag
   */
  direction: 'vertical' | 'horizontal';
  /**
   * 垂直锚点中，指示器距离顶部的宽度
   *
   * @type {number}
   * @memberof IAnchorBag
   */
  indicatorTop: number;
  /**
   * 垂直锚点中，指示器的高度
   *
   * @type {number}
   * @memberof IAnchorBag
   */
  indicatorHeight: number;
  /**
   * 注册锚点，通常在组件mount后执行
   *
   * @memberof IAnchorBag
   */
  registerLink: (link: string) => void;
  /**
   * 注销锚点，通常在组件卸载时执行
   *
   * @memberof IAnchorBag
   */
  unregisterLink: (link: string) => void;
  /**
   * 获取自定义容器节点
   *
   * @memberof IAnchorBag
   */
  getContainer: (node: HTMLElement | null) => void;
  /**
   * 设置activeLink的值
   *
   * @memberof IAnchorBag
   */
  setCurrentActiveLink: (link: string) => void;
  /**
   * 根据当前页面的滚动情况，得到计算后处于激活态的锚点
   *
   * @memberof IAnchorBag
   */
  getCurrentAnchor: (offsetTop: number, bounds: number) => string;
  /**
   * 滚动到指定锚点
   *
   * @memberof IAnchorBag
   */
  handleScrollTo: (link: string) => void;
  /**
   * 处理滚动事件的函数
   *
   * @memberof IAnchorBag
   */
  handleScroll: () => void;
  /**
   * 更新内部indicator指示器的位置
   *
   * @memberof IAnchorBag
   */
  updateIndicator: (top: number, height: number) => void;
}
