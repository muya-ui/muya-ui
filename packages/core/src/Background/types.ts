import React from 'react';

import { IBackgroundColor } from '@muya-ui/theme-light';

export interface IBackgroundPureProps {
  /**
   * 支持通用背景的几种模式
   * @default 'global'
   */
  type?: keyof IBackgroundColor;
  /**
   * 节点设置
   * @default 'div'
   */
  component?: 'div' | 'span';
  /** css display */
  display?: 'block' | 'flex' | 'inline-block' | 'inline-flex' | React.CSSProperties['display'];
  /**
   * 设置具体的背景色，酌情使用
   */
  backgroundColor?: string;
  /**
   * 文字颜色，酌情使用
   */
  color?: string;
  /** 宽 */
  width?: string | number;
  /** 高 */
  height?: string | number;
}

export type IBackgroundProps = IBackgroundPureProps &
  React.HTMLAttributes<HTMLDivElement | HTMLSpanElement>;
