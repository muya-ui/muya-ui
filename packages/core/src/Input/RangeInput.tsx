import React, { Children, useRef, useMemo } from 'react';

import { useForkRef } from '@muya-ui/utils';
import { ClearIcon as DefaultClearIcon } from '@muya-ui/theme-light';

import ClearIconWrapper from '../styled/components/ClearIconWrapper';
import addPx from '../utils/addPx';
import memoForwardRef from '../utils/memoForwardRef';
import mergeStyleItem from '../utils/mergeStyleItem';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import { ExtraNodes, StyledInput, StyledInputWrapper } from './styled';
import { IRangeInputElement, IRangeInputProps, IRangeInputStyleKeys } from './types';
import { ICustomStylePropMap } from '../types';
import useRangeInput from './useRangeInput';

const RangeInput = memoForwardRef<IRangeInputElement, IRangeInputProps>((props, ref) => {
  const {
    size = 'm',
    disabled,
    status,
    prefixNode,
    suffixNode,
    middleNode,
    allowClear,
    clearReplace,
    width,
    height,
    onMouseEnter,
    onMouseLeave,
    className,
    style,
    styles,

    // input
    value,
    defaultValue,
    placeholder = [],

    autoFocus = false,
    onFocus,
    onBlur,
    onClear,
    onClick,
    onChange,
    onKeyDown,
    onPressEnter,
    focusWhenClear = true,
    inputProps,
    ...other
  } = props;
  const theme = useTheme();
  const defaultStyles = useMemo<ICustomStylePropMap<IRangeInputStyleKeys>>(
    () => ({
      inputWrapper: '',
      headInput: '',
      tailInput: '',
      clearWrapper: '',
      nodeWrapper: '',
      nodeDivider: '',
    }),
    [],
  );
  const innerStyles = useStyles('range-input', defaultStyles, styles);

  const inputWrapperStyleItem = useMemo(
    () =>
      mergeStyleItem(
        {
          className,
          style,
        },
        innerStyles.inputWrapper,
      ),
    [className, innerStyles.inputWrapper, style],
  );

  const {
    finalValue,
    focus,
    entered,
    leftInputRef,
    rightInputRef,

    handleFocus,
    handleBlur,
    handleClick,
    handleLeftChange,
    handleRightChange,
    handleClear,
    handleMouseEnter,
    handleMouseLeave,
    handleKeyDown,
  } = useRangeInput(props);
  const innerRef = useRef<IRangeInputElement>(null);
  const more = useRef<Partial<IRangeInputElement>>({
    focus(position: 0 | 1 = 0, option?: FocusOptions) {
      if (position === 1) {
        rightInputRef.current!.focus(option);
      } else {
        leftInputRef.current!.focus(option);
      }
    },
    blur(position: 0 | 1 = 0) {
      if (position === 1) {
        rightInputRef.current!.blur();
      } else {
        leftInputRef.current!.blur();
      }
    },
  });
  const handleRef = useForkRef<IRangeInputElement>(ref, innerRef, more.current);

  const innerSuffixNode: React.ReactNode[] = [];
  Children.forEach(suffixNode, child => {
    innerSuffixNode.push(child);
  });
  const [leftValue, rightValue] = finalValue;
  // 清除输入框功能
  if (allowClear && (leftValue || rightValue) && entered && !disabled) {
    const {
      typography: {
        spec: { fontSize },
      },
      components: { Input: token },
    } = theme;
    const ClearIcon = token.clearIcon || DefaultClearIcon;
    const clearNode = (
      <ClearIconWrapper
        key="clear"
        $fontSize={fontSize[token.fontLevel[size!]]}
        onClick={handleClear}
        {...innerStyles.clearWrapper}
      >
        <ClearIcon />
      </ClearIconWrapper>
    );
    if (clearReplace) {
      innerSuffixNode.shift();
    }
    innerSuffixNode.unshift(clearNode);
  }

  const inputWidth = width !== undefined ? width : theme.components.Input.rangeInputWidth[size];

  return (
    <StyledInputWrapper
      theme={theme}
      disabled={disabled}
      focus={focus}
      $width={addPx(inputWidth)}
      $height={addPx(height)}
      size={size}
      onClick={handleClick}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      hasError={status === 'error'}
      ref={handleRef as (el: HTMLDivElement) => void}
      {...inputWrapperStyleItem}
      {...other}
    >
      <ExtraNodes
        $type="prefix"
        theme={theme}
        size={size}
        nodes={prefixNode}
        $nodeStyleItem={innerStyles.nodeWrapper}
        $dividerStyleItem={innerStyles.nodeDivider}
      />
      <StyledInput
        value={leftValue}
        disabled={disabled}
        theme={theme}
        onFocus={handleFocus}
        onBlur={handleBlur}
        onChange={handleLeftChange}
        onKeyDown={handleKeyDown}
        autoFocus={autoFocus}
        placeholder={placeholder[0]}
        size={size as any}
        ref={leftInputRef}
        {...inputProps}
      />
      <ExtraNodes
        $type="suffix"
        theme={theme}
        size={size}
        nodes={middleNode}
        full
        $nodeStyleItem={innerStyles.nodeWrapper}
        $dividerStyleItem={innerStyles.nodeDivider}
      />
      <StyledInput
        value={rightValue}
        disabled={disabled}
        theme={theme}
        onFocus={handleFocus}
        onBlur={handleBlur}
        onChange={handleRightChange}
        onKeyDown={handleKeyDown}
        autoFocus={autoFocus}
        size={size as any}
        placeholder={placeholder[1]}
        ref={rightInputRef}
        {...inputProps}
      />
      <ExtraNodes
        $type="suffix"
        theme={theme}
        size={size}
        nodes={innerSuffixNode}
        $nodeStyleItem={innerStyles.nodeWrapper}
        $dividerStyleItem={innerStyles.nodeDivider}
      />
    </StyledInputWrapper>
  );
});

export default RangeInput;
