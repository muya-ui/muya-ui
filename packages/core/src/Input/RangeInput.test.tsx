import React from 'react';
import renderer from 'react-test-renderer';
import mockConsole from 'test/utils/mockConsole';

import { cleanup, fireEvent, render } from '@testing-library/react';

import RangeInput from './RangeInput';
import { IRangeInputElement } from './types';

describe('RangeInput', () => {
  beforeAll(() => {
    mockConsole.restoreError();
    mockConsole.mockError();
  });

  afterAll(() => {
    mockConsole.restoreError();
  });

  afterEach(() => {
    cleanup();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <>
          <RangeInput />
          <RangeInput disabled />
          <RangeInput value={['ss', 'sss']} />
          <RangeInput defaultValue={['ss', 'ss']} />
          <RangeInput middleNode="to" prefixNode="s" suffixNode="ss" />
          <RangeInput
            middleNode={['ss', '', 'ss']}
            prefixNode={['ss', '', 'ss']}
            suffixNode={['ss', '', 'ss']}
          />
        </>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ref correctly', () => {
    const inputRef = React.createRef<IRangeInputElement>();
    const { container } = render(<RangeInput ref={inputRef} />);
    inputRef.current!.focus(0);
    const [leftInput, rightInput] = container.querySelectorAll('input');
    expect(document.activeElement).toBe(leftInput);
    inputRef.current!.blur(0);
    expect(document.activeElement).toBe(document.body);
    inputRef.current!.focus(1);
    expect(document.activeElement).toBe(rightInput);
    inputRef.current!.blur(1);
    expect(document.activeElement).toBe(document.body);
  });

  test('allowClear', () => {
    const { container } = render(<RangeInput defaultValue={['left', 'right']} allowClear />);
    const inputWrapper = container.firstElementChild;
    fireEvent.mouseEnter(inputWrapper!);
    const clearIconWrapper = container.querySelector(`[class*='ClearIconWrapper']`);
    fireEvent.click(clearIconWrapper!);
    const [leftInput, rightInput] = container.querySelectorAll('input');
    expect(leftInput.getAttribute('value')).toBe('');
    expect(rightInput.getAttribute('value')).toBe('');
  });
});
