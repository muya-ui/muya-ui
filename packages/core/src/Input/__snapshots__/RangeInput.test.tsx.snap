// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput should render correctly 1`] = `
Array [
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
}

.c0:hover {
  border-color: rgba(209,209,209,1);
}

.c1 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c1::selection {
  background: rgba(26,122,248,0.16);
}

.c1::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c1:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::placeholder {
  color: rgba(204,204,204,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
  </div>,
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
  background: rgba(250,250,250,1);
  cursor: not-allowed;
}

.c0 input,
.c0 svg {
  cursor: not-allowed;
}

.c1 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(221,221,221,1);
  -webkit-text-fill-color: currentColor;
  opacity: 1;
}

.c1::selection {
  background: rgba(26,122,248,0.16);
}

.c1::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c1:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::placeholder {
  color: rgba(204,204,204,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    disabled={true}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <input
      autoFocus={false}
      className="c1"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <input
      autoFocus={false}
      className="c1"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
  </div>,
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
}

.c0:hover {
  border-color: rgba(209,209,209,1);
}

.c1 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c1::selection {
  background: rgba(26,122,248,0.16);
}

.c1::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c1:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::placeholder {
  color: rgba(204,204,204,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value="ss"
    />
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value="sss"
    />
  </div>,
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
}

.c0:hover {
  border-color: rgba(209,209,209,1);
}

.c1 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c1::selection {
  background: rgba(26,122,248,0.16);
}

.c1::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c1:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c1::placeholder {
  color: rgba(204,204,204,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value="ss"
    />
    <input
      autoFocus={false}
      className="c1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value="ss"
    />
  </div>,
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
}

.c0:hover {
  border-color: rgba(209,209,209,1);
}

.c2 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c2::selection {
  background: rgba(26,122,248,0.16);
}

.c2::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c2:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2::placeholder {
  color: rgba(204,204,204,1);
}

.c1 {
  margin-right: 4px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c4:hover {
  color: rgba(187,187,187,1);
}

.c4:active {
  color: rgba(153,153,153,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <div
      className="c1 muya-range-input-node-wrapper"
      size="m"
    >
      s
    </div>
    <input
      autoFocus={false}
      className="c2"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <div
      className="c3 muya-range-input-node-wrapper"
      size="m"
    >
      to
    </div>
    <div
      className="c4 muya-range-input-node-wrapper"
      size="m"
    />
    <input
      autoFocus={false}
      className="c2"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <div
      className="c3 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
  </div>,
  .c0 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 320px;
}

.c0:hover {
  border-color: rgba(209,209,209,1);
}

.c4 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c4::selection {
  background: rgba(26,122,248,0.16);
}

.c4::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c4:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::placeholder {
  color: rgba(204,204,204,1);
}

.c3 {
  margin-right: 4px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c7:hover {
  color: rgba(187,187,187,1);
}

.c7:active {
  color: rgba(153,153,153,1);
}

.c1 {
  margin-right: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-right: 12px;
  width: 1px;
  height: 12px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: rgba(238,238,238,1);
}

.c6 {
  margin-left: 12px;
  width: 1px;
  height: 12px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: rgba(238,238,238,1);
}

<div
    className="c0 muya-range-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
  >
    <div
      className="c1 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
    <div
      className="c2 muya-range-input-node-divider"
      size="m"
    />
    <div
      className="c3 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
    <input
      autoFocus={false}
      className="c4"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <div
      className="c5 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
    <div
      className="c6 muya-range-input-node-divider"
      size="m"
    />
    <div
      className="c5 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
    <div
      className="c7 muya-range-input-node-wrapper"
      size="m"
    />
    <input
      autoFocus={false}
      className="c4"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
    <div
      className="c5 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
    <div
      className="c6 muya-range-input-node-divider"
      size="m"
    />
    <div
      className="c5 muya-range-input-node-wrapper"
      size="m"
    >
      ss
    </div>
  </div>,
]
`;
